tests/cupy_tests/cuda_tests/test_memory.py

```diff
@@ -1342,3 +1342,43 @@ def test_set_limit_fraction(self):
 
         with self.assertRaises(ValueError):
             self.pool.set_limit(fraction=1.1)
+
+
+class TestExternalAllocator(unittest.TestCase):
+    def setUp(self):
+        self.pool = memory.MemoryPool(memory._malloc)
+        self.old_pool = cupy.get_default_memory_pool()
+        memory.set_allocator(self.pool.malloc)
+
+    def tearDown(self):
+        self.pool.free_all_blocks()
+        memory.set_allocator(self.old_pool.malloc)
```

Comment:
I think we don't wanna use `unittest` anymore. I didn't test but this should work:
```suggestion


@pytest.fixture()
def ext_pool():
    old_pool = cupy.get_default_memory_pool()
    new_pool = memory.MemoryPool(memory._malloc)
    memory.set_allocator(new_pool.malloc)
    yield new_pool
    new_pool.free_all_blocks()
    memory.set_allocator(old_pool.malloc)


class TestExternalAllocator:
```
and pass `ext_pool` to all test methods as an argument (ex: `def test_malloc(self, ext_pool):`).